appveyor.sh

```diff
@@ -15,7 +15,7 @@ then
   ./createcoveragereport.sh
   if [[ -f "coverage/coverage-filtered.xml" ]]
   then
-    codecov -f "coverage/coverage-filtered.xml"
+    codecov -f "coverage/coverage-filtered.xml" --flag unittests
```

Comment:
As a thought, I wonder whether we should put this logic into createcoveragereport - it would certainly simplify other scripts. Not something for this PR, but let me know if you think it's a good idea.

Reply:
Definitely, one always comes after the other, and actually on createcoveragereport there's a TODO that says something like "upload report".
I'll send you another PR with that change.